### ISSUE TITLE
"ALTER IGNORE TABLE" command no longer works in Ubuntu 16.04 LTS on second pass of "$LJHOME/bin/upgrading/update-db.pl -r --innodb"

### ISSUE THREAD

**Issue by Sophira** [dreamwidth/dw-free#1823]
When creating a new Dreamwidth installation on Ubuntu 16.04 LTS (and working around #1822 by using "0000-01-01 00:00:00" as a default value and doing a very sloppy find-and-replace job to update the code), the following error is seen on the second pass of `$LJHOME/bin/upgrading/update-db.pl -r --innodb`:

>     ALTER IGNORE TABLE moodthemedata DROP KEY moodthemeid;
>     # Running...
>     #  ERROR: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'IGNORE TABLE moodthemedata DROP KEY moodthemeid' at line 1

This error doesn't seem to be related to my (admittedly sloppy and not submittable) workaround for issue #1822, so I'm submitting this as a separate issue.

Output of `mysql -V`:

>     mysql  Ver 14.14 Distrib 5.7.12, for Linux (x86_64) using  EditLine wrapper

**Comment by kareila** [comment]
Does the error go away if you remove KEY (moodthemeid) from the table declaration before initializing the database? That seems like it would preempt the execution of the statement.

**Comment by Sophira** [comment]
Yes, it does. I'll submit a pull request for this as soon as I can.

By the way, should I be submitting each Ubuntu 16.04 LTS problem in a separate issue, or is it fine to put them in separate issues? (Because there's at least one more issue that turns up after this one, too.)

**Comment by wohali** [comment]
I'm working on updating the documentation for a DW scratch install and just ran headlong into this. I can confirm this bug is unrelated to the use of strict mode, and fails regardless.

@Sophira do you have a branch with cleanup work on this prepped or should I attempt to fix it myself and submit a PR?

**Comment by Sophira** [comment]
I don't have it done yet - I was going to make a pull request but haven't done so yet as I was going to wait until I had finished running through the installation of the Dreamhack code on a VM and then do the necessary updates.

You're welcome to make one, though! @Kareila outlined a fix above, and it involves editing `bin/upgrading/update-db-general.pl`; search for `CREATE TABLE moodthemedata` to find the table definition, then as @Kareila says, remove the "KEY (moodthemeid)," line. :)

**Comment by kareila** [comment]
Go for it, @wohali! Make sure you get your CLA submitted, though - I can't merge your pull request until you do.

**Comment by wohali** [comment]
Just submitted my CLA. I'll try and get a PR done shortly.

**Comment by wohali** [comment]
PR is up, not sure why it didn't get flagged here. 
https://github.com/dreamwidth/dw-free/pull/1859

**Comment by kaberett** [comment]
The pull req needs to include "Fixes #XXXX" in a line by itself for the magical cross-linking to work :-)

**Comment by wohali** [comment]
Got it! I thought having the fixes #### in the title would be enough, guess not :person_frowning:

**Comment by kareila** [comment]
Ugh, the auto-assigning also doesn't work if you're not a member of the organization.  You're supposed to get an invitation to join when your CLA is filed - if we dropped the ball on that, let us know and I'll try to get it resent?

**Comment by rahaeli** [comment]
My fault -- I got wohali's CLA, but it looks like I didn't send the invite when I did.

**Comment by wohali** [comment]
Do I need to assign this to myself somehow still?

**Comment by rahaeli** [comment]
If you've accepted the invite to be part of the org, you can comment on an issue with the word "c.l.a.i.m" (without the periods) and it will assign it to you!
